src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -559,6 +580,134 @@ export abstract class VerifiableCredentialsServiceBase {
       credential.expirationDate = params.expirationDate.toISOString();
     }
 
+    if (params.credentialStatus) {
+      credential.credentialStatus = params.credentialStatus;
+    }
+
     return credential;
   }
+
+  /**
+   * Revoke given verifiable credential with StatusList2021.
+   *
+   * ```typescript
+   * await verifiableCredentialsService.revokeCredential(credential);
+   * ```
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject>} credential verifiable credential
+   * @return StatusList2021Credential
+   */
+  public async revokeCredential(
+    credential: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<StatusList2021Credential> {
+    const statusListUnsignedCredential =
+      await this._cacheClient.initiateCredentialStatusUpdate(credential);
+
+    const proofOptionsObject = {
+      verificationMethod: `${this._signerService.didHex}#controller`,
+      proofPurpose: 'assertionMethod',
+      eip712Domain: {
+        primaryType: 'VerifiableCredential',
+        domain: {},
+        messageSchema: statusList2021CredentialEIP712Types,
+      },
+    };
+
+    const keyType = {
+      kty: 'EC',
+      crv: 'secp256k1',
+      alg: 'ES256K-R',
+      key_ops: ['signTypedData'],
+    };
+
+    const stringifyCredential = JSON.stringify(statusListUnsignedCredential);
+    const preparedVC = await this.prepareIssueCredential(
+      stringifyCredential,
+      JSON.stringify(proofOptionsObject),
+      JSON.stringify(keyType)
+    );
+    const preparation = JSON.parse(preparedVC);
+
+    const typedData = preparation.signingInput;
+    if (!typedData || !typedData.primaryType) {
+      throw new Error('Expected EIP-712 TypedData');
+    }
+
+    delete typedData.types['EIP712Domain'];
+
+    const signature = await this._signerService.signTypedData(
+      typedData.domain,
+      typedData.types,
+      typedData.message
+    );
+
+    const signedCredential = await this.completeIssueCredential(
+      stringifyCredential,
+      preparedVC,
+      signature
+    );
+
+    const statusList2021Credential = JSON.parse(
+      signedCredential
+    ) as StatusList2021Credential;
+
+    return await this._cacheClient.persistCredentialStatusUpdate(
+      statusList2021Credential
+    );
+  }
+
+  /**
+   * Revoke given verifiable credential with StatusList2021.
+   *
+   * ```typescript
+   * await verifiableCredentialsService.isRevoked(credential);
+   * ```
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject>} credential verifiable credential
+   * @return true if credential is revoked
+   */
+  public async isRevoked(
+    credential: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<boolean> {
+    const statusListCredential =
```

Comment:
Is this method equivalent to return (await revocationDetails(credential)) !== null?